packages/application/src/shell.ts

```diff
@@ -103,12 +104,29 @@ export class NotebookShell extends Widget implements JupyterFrontEnd.IShell {
     return this._menuWrapper;
   }
 
+  /**
+   * Get the status of main area.
+   */
+  get mainIsTabPanel(): Boolean {
```

Comment:
Need to think about this, but at first glance trying to know what a page or a set of extensions is trying to add sounds a bit odd.

The shell should be generic enough, for example we could imaging other custom application reusing `NotebookShell` but without the tree view (or a different one not using a `TabPanel`).

Reply:
I agree that this solution is not generic enough.

> Also for more context the `NotebookShell` was initially thought as only allowing 1 widget in the main area on purpose.

Is that still relevant as there is already the tree view that would allow several widgets in main area ?

Indeed if the tree view will be the only one needing several widgets, the better way might be to expose its tab panel as said in #6410. Would the idea be to use token ?

Otherwise, another way could be to use a dock panel like in JupyterLab, but limited to one widget by default. That would keep the initial context but allows several widgets in main area if necessary.

Reply:
> Is that still relevant as there is already the tree view that would allow several widgets in main area ?

Good point. For the tree page it would make sense for widgets to be added to the tab panel. And would be natural to extension authors using `app.shell.add(widget, 'main')`.


> Would the idea be to use token ?

Yes the idea would have been to require the token to be able to add new widgets to the tab panel. Thinking more about it this might be simpler to implement, although maybe less straighforward to extension authors who would somehow need to create a dedicated plugin for this.